framework/src/modules/chain/bft/bft.js

```diff
@@ -14,273 +14,135 @@
-	 * @return {BFT}
+	 * @param {Object} storage - Storage component instance
+	 * @param {Object} logger - Logger component instance
+	 * @param {integer} activeDelegates - Number of delegates
+	 * @param {integer} startingHeight - The height at which BFT consensus initialize
 	 */
-	addBlockHeader(blockHeader) {
-		// Validate the schema of the header
-		// To spy exported function in same module we have to call it as this
-		exportedInterface.validateBlockHeader(blockHeader);
-
-		// Verify the integrity of the header with chain
-		this.verifyBlockHeaders(blockHeader);
-
-		// Add the header to the list
-		this.headers.add(blockHeader);
-
-		// Update the pre-votes and pre-commits
-		this.updatePreVotesPreCommits(blockHeader);
-
-		// Update the pre-voted confirmed and finalized height
-		this.updatePreVotedAndFinalizedHeight();
+	constructor({ storage, logger, activeDelegates, startingHeight }) {
+		this.finalityManager = null;
+
+		this.logger = logger;
+		this.storage = storage;
+		this.constants = {
+			activeDelegates,
+			startingHeight,
+		};
 
-		return this;
+		this.BlockEntity = this.storage.entities.Block;
+		this.ChainMetaEntity = this.storage.entities.ChainMeta;
 	}
 
 	/**
-	 * Update pre-votes and pre-commits in reference to particular block header
+	 * Initialize the BFT module
 	 *
-	 * @param {BlockHeader} lastBlockHeader
-	 * @return {undefined}
+	 * @return {Promise<void>}
 	 */
-	updatePreVotesPreCommits(lastBlockHeader) {
-		// Update applies particularly in reference to last block header in the list
-		const header = lastBlockHeader || this.headers.last;
-
-		// If delegate forged a block with higher or same height previously
-		// that means he is forging on other chain and we don't count any
-		// pre-votes and pre-commits from him
-		if (header.maxHeightPreviouslyForged >= header.height) {
-			return;
-		}
-
-		// Get delegate public key
-		const delegatePublicKey = header.delegatePublicKey;
-
-		// Load or initialize delegate state in reference to current BFT block headers
-		const delegateState = this.state[delegatePublicKey] || {
-			maxPreVoteHeight: 0,
-			maxPreCommitHeight: 0,
-		};
-
-		// Get first block of the round when delegate was active
-		const delegateMinHeightActive =
-			(header.activeSinceRound - 1) * this.activeDelegates + 1;
-
-		// If delegate is new then first block of the round will be considered
-		// if it forged before then we probably have the last commit height
-		// delegate can't pre-commit and block before above mentioned conditions
-		const minPreCommit = Math.max(
-			delegateMinHeightActive,
-			delegateState.maxPreCommitHeight + 1
+	async init() {
+		this.finalityManager = await this._initFinalityManager();
+		const finalizedHeight = this.finalityManager.finalizedHeight;
+		const lastBlockHeight = await this._getLastBlockHeight();
+
+		const loadFromHeight = Math.max(
+			finalizedHeight,
+			lastBlockHeight - this.constants.activeDelegates * 2,
+			this.constants.startingHeight
```

Comment:
What does this constant signify?

Reply:
Its the migration height for the network where we introduce the BFT. That probably will the height to introduce block version 2